yellowbrick/classifier/base.py

```diff
@@ -33,119 +37,159 @@
-            will be fit when the visualizer is fit, otherwise, the estimator will not be
-            modified. If "auto" (default), a helper method will check if the estimator
-            is fitted before fitting it again.
-
-        kwargs : dict
-            Keyword arguments that are passed to the base class and may influence
-            the visualization as defined in other Visualizers. Optional keyword
-            arguments include:
-
-        .. todo:: Finish documenting class.
-        .. todo:: accept as input ``classes``, as all visualizers need this.
-        """
         # A bit of type checking
-        if not isclassifier(model):
+        if not force_model and not isclassifier(model):
             raise YellowbrickTypeError(
                 "This estimator is not a classifier; "
                 "try a regression or clustering score visualizer instead!"
             )
 
-        # Convert to array if necessary to match estimator.classes_
-        if classes is not None:
-            classes = np.array(classes)
-
-        # Set up classifier score visualization properties
-        if classes is not None:
-            n_colors = len(classes)
-        else:
-            n_colors = None
-
-        self.colors = color_palette(kwargs.pop("colors", None), n_colors)
-        self.classes_ = classes
-
         # Initialize the super method.
         super(ClassificationScoreVisualizer, self).__init__(
-            model, ax=ax, fig=fig, **kwargs
+            model, ax=ax, fig=fig, is_fitted=is_fitted, **kwargs
         )
 
+        self.set_params(classes=classes, encoder=encoder, force_model=force_model)
+
     @property
-    def classes_(self):
+    def colors(self):
         """
-        Proxy property to smartly access the classes from the estimator or
-        stored locally on the score visualizer for visualization.
+        Returns ``_colors`` if it exists, otherwise computes a categorical color
+        per class based on the matplotlib color cycle. If the visualizer is not
+        fitted, raises a NotFitted exception.
+
+        If subclasses require users to choose colors or have specialized color
+        handling, they should set ``_colors`` on init or during fit.
+
+        Notes
+        -----
+        Because this is a property, this docstring is for developers only.
         """
-        if self.__classes is None:
-            try:
-                return self.estimator.classes_
-            except AttributeError:
-                return None  # TODO: raise NotFittedError instead of returning None
-        return self.__classes
+        if not hasattr(self, "_colors"):
+            if not hasattr(self, "classes_"):
+                raise NotFitted("cannot determine colors before fit")
 
-    @classes_.setter
-    def classes_(self, value):
-        self.__classes = value
+            # TODO: replace with resolve_colors
+            self._colors = color_palette(None, len(self.classes_))
```

Comment:
If we replace this with resolve colors we'll need to add colors/cmap arguments - but none of the current visualizers use class-specific colors (though ClassPredError, ROCAUC and PRCurve could benefit from this)

Reply:
👌